### ISSUE TITLE
Importing Python files from project to entry script of ML service

### ISSUE THREAD

**Issue by Konstantina-Lazaridou** [Azure/MachineLearningNotebooks#1131]
My project structure for deploying a classifier service in Azure is the following:
```
project
      src
          __init__.py
          register.sh
          deploy.sh
          deployment_config.json
          inference_config.json
          entry_script.py
          entry_script_utils.py
          ml_scripts/
              some_script.py
 
```

In my entry script, specifically in `run`,  I include some functions for pre and post processing, which I'd rather keep in `entry_script_utils.py` instead. It's cleaner and it's also easier to test locally. 

The code in the entry script is the following:
```

def init():
    global model
    model = myclassifier() 

def run(data: str):
    def util_1() ...
    def util_2() ...
    def util_3() ...
  ....
  # use utils
  # use model
  # use utils
  return result
```

When moving e.g. `util_1` to `entry_script_utils`, I try to access it within my entry script with 
`from src.entry_script_utils import util_1` or `from .entry_script_utls import util_1`, 
but none works. The "entry_script_utils" module is unrecognized. I tried `sys.path.append(".")`, but it didn't help.

Also, if I use relative imports like `from .ml_scripts import some_script`, I also get an error. The error message is:
`attempted relative import with no known parent package`.  I tried `sys.path.append(".ml_scripts")`, but it also didn't work. 

When I registered my classification model in Azure, I also added some code in the directory of the model. So, I was hoping I can access these scripts now. I tried adding my `entry_script_utils.py` there and then access like that 
```
    model_path = os.path.join(os.getenv('AZUREML_MODEL_DIR'), '/model/')
    sys.path.append(model_path)
    from entry_script_utils import util_1
```
but `entry_script_utils` is not found. 

I am following [this](https://github.com/Azure/MachineLearningNotebooks/issues/833) issue, which is similar to mine. As proposed, I added 
`import os, sys
sys.path.append(os.path.dirname(os.path.realpath(__file__)))` 
as a workaround at the beginning of my `entry_script.py`, but nothing changed. 

The second suggestion was to add the "source_directory" attribute in the inference configuration, i.e., 
`"entryScript": "entry_script.py",
    "source_directory": "."`,
which I ran while deploying my model from within `src/`, but that also didn't help. Instead of ".", I used my absolute path of the "src" folder as a test, but that didn't change the import problem.

Is there any way to have multiple scripts work together with the `entry_script`? I understand that the purpose of the entry script is to just consume an ML model, but in some cases, pre/post processing is necessary, which makes having a very long `run` function not sustainable.

**Comment by shivanissambare** [comment]
@jonringer can you please help?

**Comment by jonringer** [comment]
Similar to https://github.com/Azure/MachineLearningNotebooks/issues/833 this is caused by the current working directory being set to the parent of the source directory on model deployment. This will be fixed in the next SDK release (azureml-sdk 1.14) where the current working directory will be set to the `source_directory` if present in the inference config. By default, the python interpreter will also add the current working directory to `sys.path` which is why this works locally.

### To fix your issue immediately
You need to mention `src/` as the source_directory, and then in `entry_script.py`, you can't make assumptions about where the `entry_script.py` or any other script is in relation to the current working directory.
An example would be:
```python
    # entry_script.py
    # find directory in which entry_script.py exists
    SCRIPT_DIRECTORY = os.path.dirname(os.path.realpath(__file__))
    # add to import paths
    sys.path.append(SCRIPT_DIRECTORY)
    from entry_script_utils import util_1
```
and set the source_directory to `src/`, unfortunately, I'm not aware of a good way to determine this through json
```json
{
    "entryScript" : "entry_script.py",
    "sourceDirectory" : "/path/to/project/src",
    "envrionment" : {
        ....
    }
}
```
If you have `jq` installed, you could do something like this:
```bash
# deploy.sh
# find parent directory of deploy.sh
cwd="$(dirname $(realpath ${BASH_SOURCE[0]}))"
# set source_directory to point to the parent directory
jq ".sourceDirectory = \"${cwd}\"" "$cwd"/inferenceconfig.json
```

### Other thoughts
> The second suggestion was to add the "source_directory" attribute in the inference configuration

The `source_directory` also needs to be mentioned, as the default behavior is to only deploy the `entry_script.py`, and the source_directory allows for all files in a given directory to be deployed in a model deploy. (NOTE: the entry_script needs to be a relative path from the `source_directory`, otherwise deployment will fail, but the SDK will validate that the paths are correct on the client sdk side)

So in your example:
```
project
      src
          __init__.py
          register.sh
          deploy.sh
          deployment_config.json
          inference_config.json
          entry_script.py
          entry_script_utils.py
          ml_scripts/
              some_script.py
```
you need
```
   "entryScript" = "entry_script.py",
   "sourceDirectory" = ".",
```
Although resolving the source directory would be best, to avoid further complications due to implicit current working directory behavior. (see jq example above)

### As a side note:
Once azureml-sdk 1.14 is released, you should be able to update your sdk and redeploy the models with `source_directory` pointed to `src/`, and they "should" work with relative paths. Assuming that all relative paths are relative to `src/`.

**Comment by shivanissambare** [comment]
Thank you @jonringer 
I am closing this issue
please re-open if you feel your problem isn't solved
